### PR TITLE
Sequential feature extractor remove_unused

### PR DESCRIPTION
**Pull request by yuyu2172**
Merge after #347 .

I found that it is beneficial to have a method that chops off unnecessary part from a feature extractor.
For instance, an optimizer hook is called for all parameters even if they are not used in a computational graph.

### CODE REVIEW

**Comment by Hakuyume** [comment]
I merged #347. Could you merge master branch to this PR?

**Comment by yuyu2172** [comment]
Merge after https://github.com/chainer/chainercv/pull/349

File: chainercv/links/model/sequential_feature_extractor.py [hunk 1]
```diff
@@ -111,21 +106,21 @@ def feature_names(self, feature_names):
...
-            for name in list(self.all_feature_names):
-                if name not in keep_feature_names:
-                    delattr(ret, name)
-            return ret
+    def remove_unused(self):
+        """Delete all features that are not needed for the forward pass.
+
+        """
+        if self._feature_names is None:
+            feature_names = (self.all_feature_names[-1],)
```
Comment (Hakuyume) [inline chainercv/links/model/sequential_feature_extractor.py]: If `_feature_names` is `None`, we can simply use `return` to skip this method.

File: chainercv/links/model/sequential_feature_extractor.py [hunk 2]
```diff
@@ -111,21 +106,21 @@ def feature_names(self, feature_names):
...
-            return getattr(self, self.all_feature_names[index])
-        elif isinstance(index, slice):
-            ret = self.copy()
-            keep_feature_names = self.all_feature_names[index]
-            for name in list(self.all_feature_names):
-                if name not in keep_feature_names:
-                    delattr(ret, name)
-            return ret
+    def remove_unused(self):
+        """Delete all features that are not needed for the forward pass.
```
Comment (Hakuyume) [inline chainercv/links/model/sequential_feature_extractor.py]: This function removes layers (function or link). I prefer `Delete all layers`.

**Pull request merged**